app/models/manageiq/providers/hawkular/inventory/parser/middleware_domains.rb

```diff
@@ -35,12 +35,16 @@ def fetch_domain_availabilities
       collection = persister.middleware_domains
       fetch_availabilities_for(collector.domains, collection, collection.model_class::AVAIL_TYPE_ID) do |domain, availability|
         domain.properties['Availability'] =
-          process_domain_availability(availability.try(:[], 'data').try(:first))
+          process_domain_availability(availability)
       end
     end
 
     def process_domain_availability(availability = nil)
-      process_availability(availability, 'up' => 'Running', 'down' => 'Stopped')
+      if availability.first['value'] && availability.first['value'][1] == '1'
```

Comment:
I can see that now there is no chance to that `availability` comes as `nil`. This line will cause an exception if that happens. If this is now unsupported, perhaps we shouldn't `yield` in the `fetch_availabilities_for` method if the availability metric is not there... But we will removing the "unknown" case. Not sure if you designed that way.

Reply:
Probably I miss something. I'm going to review it in master.
Thanks for the point.

Reply:
Yes, I confirm I removed the 'unknown' case, I'm going to fix that.
One question, what is the expected behaviour when the whole provider is down ?
At the moment we only notify that there is not connectivity with the provider, but dependent objects maintain the status.
Should they set up to 'unknown'/'down' ?

Reply:
The proper setting would be 'unknown', but if we don't want to deal with that I think the last know state is appropriate.  It's quite possible the monitored servers are running.  If we notify that the provider is down, that is probably good enough.

Reply:
Yes, I think it's not too trivial to update all entities to 'unknown' status. It would be like bypassing the refresh process flow to update the statuses. Currently, MiQ will display the error in the "Last refresh date" field of the provider page. I think that's enough for now.